ordering/__init__.py

```diff
@@ -103,6 +104,14 @@ def assert_new_item(self, item: T) -> None:
         if item in self:
             raise KeyError("Ordering {} already contains {}".format(self, item))
 
+    @classmethod
+    def assert_unique_items(cls, items: Collection[_T]) -> None:
+        if len(items) > len(set(items)):
+            d: Dict[_T, bool] = {}
+            duplicates = [i for i in items if i in d or d.setdefault(i, False)]
```

Comment:
I'm happy to go either way on that point.

Showing all the duplicate items may be more convenient, as it stops the potential cycle of back and forth of fixing one problem, just for another to pop up.

At the same time, it is possible that showing all duplicates may be overwhelming, and it doesn't match with the standard library.

Do whichever is most convenient for you.